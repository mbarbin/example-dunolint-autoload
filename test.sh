#!/bin/bash
set -e

# Test script for dunolint autoloading config feature
# Tests various invocation scenarios to verify config files are loaded correctly

DUNOLINT=dunolint
ROOT_DIR="$(pwd)"

echo "=== Test 1: Run dunolint from root directory ==="
echo "Expected: Root config should be loaded and applied"
echo
cd "$ROOT_DIR"
$DUNOLINT lint --dry-run

echo
echo "=== Test 2: Run dunolint with --below lib (from root) ==="
echo "Expected: Only check lib/, applying root config"
echo
cd "$ROOT_DIR"
$DUNOLINT lint --dry-run --below lib

echo
echo "=== Test 3: Run dunolint with --below repo/foo (from root) ==="
echo "Expected: Check repo/foo/, should load foo's config"
echo
cd "$ROOT_DIR"
$DUNOLINT lint --dry-run --below repo/foo

echo
echo "=== Test 4: Run dunolint with --below repo/bar (from root) ==="
echo "Expected: Check repo/bar/, should load bar's config"
echo
cd "$ROOT_DIR"
$DUNOLINT lint --dry-run --below repo/bar

echo
echo "=== Test 5: Run dunolint from repo/foo (foo as root) ==="
echo "Expected: Foo config as root, should apply foo-specific rules"
echo
cd "$ROOT_DIR/repo/foo"
$DUNOLINT lint --dry-run

echo
echo "=== Test 6: Run dunolint from repo/bar (bar as root) ==="
echo "Expected: Bar config as root, should apply bar-specific rules"
echo
cd "$ROOT_DIR/repo/bar"
$DUNOLINT lint --dry-run

echo
echo "=== All tests completed ==="
