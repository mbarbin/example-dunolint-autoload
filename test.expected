=== Test 1: Run dunolint from root directory ===
Expected: Root config should be loaded and applied

dry-run: Would edit file "dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name dunolint_test_root)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "lib/dune":
-1,2 +1,4
  (library
!| (name root_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

dry-run: Would edit file "repo/bar/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name bar)
  
-|(implicit_transitive_deps false)
+|(implicit_transitive_deps true)

dry-run: Would edit file "repo/bar/lib/dune":
-1,2 +1,4
  (library
!| (name bar_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

dry-run: Would edit file "repo/foo/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name foo)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "repo/foo/lib/dune":
-1,2 +1,4
  (library
-| (name wrong_name))
+| (name foo_lib)
+| (instrumentation
+|  (backend bisect_ppx)))

=== Test 2: Run dunolint with --below lib (from root) ===
Expected: Only check lib/, applying root config

dry-run: Would edit file "lib/dune":
-1,2 +1,4
  (library
!| (name root_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

=== Test 3: Run dunolint with --below repo/foo (from root) ===
Expected: Check repo/foo/, should load foo's config

dry-run: Would edit file "repo/foo/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name foo)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "repo/foo/lib/dune":
-1,2 +1,4
  (library
-| (name wrong_name))
+| (name foo_lib)
+| (instrumentation
+|  (backend bisect_ppx)))

=== Test 4: Run dunolint with --below repo/bar (from root) ===
Expected: Check repo/bar/, should load bar's config

dry-run: Would edit file "repo/bar/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name bar)
  
-|(implicit_transitive_deps false)
+|(implicit_transitive_deps true)

dry-run: Would edit file "repo/bar/lib/dune":
-1,2 +1,4
  (library
!| (name bar_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

=== Test 5: Run dunolint from repo/foo (foo as root) ===
Expected: Foo config as root, should apply foo-specific rules

Entering directory '/home/mathieu/dev/test/dunolint-autoload'
dry-run: Would edit file "dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name dunolint_test_root)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "lib/dune":
-1,2 +1,4
  (library
!| (name root_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

dry-run: Would edit file "repo/bar/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name bar)
  
-|(implicit_transitive_deps false)
+|(implicit_transitive_deps true)

dry-run: Would edit file "repo/bar/lib/dune":
-1,2 +1,4
  (library
!| (name bar_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

dry-run: Would edit file "repo/foo/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name foo)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "repo/foo/lib/dune":
-1,2 +1,4
  (library
-| (name wrong_name))
+| (name foo_lib)
+| (instrumentation
+|  (backend bisect_ppx)))

=== Test 6: Run dunolint from repo/bar (bar as root) ===
Expected: Bar config as root, should apply bar-specific rules

Entering directory '/home/mathieu/dev/test/dunolint-autoload'
dry-run: Would edit file "dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name dunolint_test_root)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "lib/dune":
-1,2 +1,4
  (library
!| (name root_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

dry-run: Would edit file "repo/bar/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name bar)
  
-|(implicit_transitive_deps false)
+|(implicit_transitive_deps true)

dry-run: Would edit file "repo/bar/lib/dune":
-1,2 +1,4
  (library
!| (name bar_lib)
!| (instrumentation
!|  (backend bisect_ppx)))

dry-run: Would edit file "repo/foo/dune-project":
-1,5 +1,5
  (lang dune 3.18)
  
  (name foo)
  
-|(implicit_transitive_deps true)
+|(implicit_transitive_deps false)

dry-run: Would edit file "repo/foo/lib/dune":
-1,2 +1,4
  (library
-| (name wrong_name))
+| (name foo_lib)
+| (instrumentation
+|  (backend bisect_ppx)))

=== All tests completed ===
